File: lambdas/websocket/connect.ts
import { APIGatewayEvent, Context, APIGatewayProxyResult } from 'aws-lambda';
import * as AWS from 'aws-sdk';

// Environment variable
const CONNECTIONS_TABLE = process.env.CONNECTIONS_TABLE || 'ConnectionsTable';

// Initialize DynamoDB DocumentClient
const dynamodb = new AWS.DynamoDB.DocumentClient();
const connectionsTable = CONNECTIONS_TABLE;

export const lambdaHandler = async (event: APIGatewayEvent, context: Context): Promise<APIGatewayProxyResult> => {
    const connectionId = event.requestContext.connectionId;
    console.log(`New connection: ${connectionId}`);

    // Extract additional information if available (e.g., userId)
    const queryStringParameters = event.queryStringParameters || {};
    const userId = queryStringParameters.userId;  // Assuming the client sends userId as a query parameter

    // Store the connection details in DynamoDB
    const connectionItem = {
        connectionId,
        userId,  // Store userId if available
        connectedAt: event.requestContext.connectedAt,  // Time of connection
    };

    try {
        // Save the connection in DynamoDB
        await dynamodb.put({
            TableName: connectionsTable,
            Item: connectionItem,
        }).promise();

        return {
            statusCode: 200,
            body: 'Connected successfully.',
        };
    } catch (error) {
        console.error('Error saving connection:', error);
        return {
            statusCode: 500,
            body: 'Failed to connect.',
        };
    }
};

File: lambdas/src/websocket/connect.py
import os
import boto3

CONNECTIONS_TABLE = os.getenv('CONNECTIONS_TABLE', 'ConnectionsTable')

dynamodb = boto3.resource('dynamodb')
connections_table = dynamodb.Table(CONNECTIONS_TABLE)

def handler(event, context):
    connection_id = event['requestContext']['connectionId']
    print(f'New connection: {connection_id}')
    
    # Extract additional information if available (e.g., userId or driverId)
    query_string_parameters = event.get('queryStringParameters', {})
    user_id = query_string_parameters.get('userId')  # Assuming the client sends userId as a query parameter
    
    # Store the connection details in DynamoDB
    connection_item = {
        'connectionId': connection_id,
        'userId': user_id,  # Store userId if available
        'connectedAt': event['requestContext']['connectedAt']  # Time of connection
    }
    
    # Save the connection in DynamoDB
    connections_table.put_item(Item=connection_item)
    
    return {
        'statusCode': 200,
        'body': 'Connected successfully.'
    }
